src/main/java/com/katalon/jenkins/plugin/ExecuteKatalonTestOpsPlan.java

```diff
@@ -175,10 +176,14 @@ private String getApiKey(String credentialsId) {
       return credentials == null ? null : credentials.getSecret().getPlainText();
     }
 
+    @RequirePOST
     public FormValidation doTestConnection(@QueryParameter("serverUrl") final String url,
                                            @QueryParameter("credentialsId") final String credentialsId) {
+      Jenkins.getActiveInstance().checkPermission(Item.CONFIGURE);
+      Jenkins.getActiveInstance().checkPermission(Item.EXTENDED_READ);
```

Comment:
Item/Configure implies Item/ExtendedRead, so this will never throw.

Reply:
Updated